### PR TITLE
tcp: check option length before d_len update

### PR DESCRIPTION
**Pull request by zhhyu7**
Signed-off-by: zhanghongyu <zhanghongyu@xiaomi.com>
Change-Id: I67eb94228c32f70eb5622426e680ed562eb9d42d

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@zhhyu7 remove the change-id from commit message.

File: net/tcp/tcp_input.c [hunk 1]
```diff
@@ -616,6 +616,17 @@ static void tcp_input(FAR struct net_driver_s *dev, uint8_t domain,
       goto drop;
     }
 
+  /* d_appdata should remove the tcp specific option field. */
+
+  if ((tcp->tcpoffset & 0xf0) > 0x50)
+    {
+      len = ((tcp->tcpoffset >> 4) - 5) << 2;
```
Comment (pkarashchenko) [inline net/tcp/tcp_input.c:623]: I do not understand this. The `len` value is unconditionally overwritten few lines below.
Maybe `len = (tcp->tcpoffset >> 4) << 2;` at line 634 should be relocated to the `else` of `if ((tcp->tcpoffset & 0xf0) > 0x50)`?

Comment (zhhyu7) [reply net/tcp/tcp_input.c]: > I do not understand this. The `len` value is unconditionally overwritten few lines below. Maybe `len = (tcp->tcpoffset >> 4) << 2;` at line 634 should be relocated to the `else` of `if ((tcp->tcpoffset & 0xf0) > 0x50)`?

They're calculated separately, so maybe they're a little redundant. Offset do not contain options length when tcp_ipvx_select, so need to skip options field when TCP_NEWDATA flag set. Should I skip options field in tcp_ipvx_select function with IPvxTCP_HDRLEN? It doesn't feel right either.

**Review by xiaoxiang781216 — APPROVED** [review]

File: net/tcp/tcp_input.c (hunk 1 above)
Comment (pkarashchenko) [reply net/tcp/tcp_input.c]: Now changes looks better. I was not saying that your initial code was wrong, but at least hard to track what is going on.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**